hack/for-go-proj.sh

```diff
@@ -19,58 +19,39 @@ set -o pipefail
 set -o nounset
 
 CONTRIB_ROOT="$(dirname ${BASH_SOURCE})/.."
-PROJECT_NAMES=(addon-resizer vertical-pod-autoscaler)
 
 if [[ $# -ne 1 ]]; then
-  echo "missing subcommand: [build|install|test]"
+  echo "missing subcommand: [cluster-autoscaler|vertical-pod-autoscaler|addon-resizer]"
   exit 1
 fi
 
-CMD="${1}"
-
-case "${CMD}" in
-  "build")
+PROJECT="${1}"
+
+case "${PROJECT}" in
+  "cluster-autoscaler")
+    pushd ${CONTRIB_ROOT}/cluster-autoscaler/
+    # TODO: #8127 - Use default analyzers set by `go test` to include `printf` analyzer.
+    # Default analyzers that go test runs according to https://github.com/golang/go/blob/52624e533fe52329da5ba6ebb9c37712048168e0/src/cmd/go/internal/test/test.go#L649
+    # This doesn't include the `printf` analyzer until cluster-autoscaler libraries are updated.
+    ANALYZERS="atomic,bool,buildtags,directive,errorsas,ifaceassert,nilfunc,slog,stringintconv,tests"
+    go test -count=1 ./... -vet="${ANALYZERS}"
+    popd
     ;;
-  "install")
+  "vertical-pod-autoscaler")
+    pushd ${CONTRIB_ROOT}/vertical-pod-autoscaler
+    go test -count=1  -race $(go list ./... | grep -v /vendor/ | grep -v vertical-pod-autoscaler/e2e | grep -v cluster-autoscaler/apis)
```

Comment:
Hm, the `cluster-autoscaler/apis` shouldn't be relevant anymore, it only applied to CA. Same with `vertical-pod-autoscaler/e2e` and `addon-resizer` below.

Reply:
I'll take this feedback and the feedback for per-project shell scripts as a follow-up.